### PR TITLE
[ZEPPELIN-599]notebook search should search paragraph title

### PR DESCRIPTION
**Pull request by ravicodder**
### What is this PR for?
Allow notebook search to search paragraph title too.

### What type of PR is it?
[Bug Fix] 
### What is the Jira issue?
[ZEPPELIN-599](https://issues.apache.org/jira/browse/ZEPPELIN-599?jql=project%20%3D%20ZEPPELIN%20AND%20status%20%3D%20Open%20AND%20text%20~%20%22Title%22)

### How should this be tested?
 You should be able to search the the note by searching queryterm in the paragraph  title.

### Screenshots (if appropriate)
Before:
screen shot 2016-04-27 at 10 33 29 pm

After:
screen shot 2016-04-27 at 10 24 14 pm

### Questions:
* Does the licenses files need update?NO
* Is there breaking changes for older versions?NO
* Does this needs documentation?NO

### CODE REVIEW

**Comment by ravicodder** [comment]
Ready for review.

File: zeppelin-web/src/app/search/result-list.controller.js [hunk 1]
```diff
@@ -16,104 +16,133 @@
 
 angular
   .module('zeppelinWebApp')
-  .controller('SearchResultCtrl', function($scope, $routeParams, searchService) {
+  .controller('SearchResultCtrl', function ($scope, $routeParams, searchService) {
 
-  var results = searchService.search({'q': $routeParams.searchTerm}).query();
+    var results = searchService.search({'q': $routeParams.searchTerm}).query();
```
Comment (Leemoonsoo) [inline zeppelin-web/src/app/search/result-list.controller.js]: I think original indent is correct.

**Comment by Leemoonsoo** [comment]
Thanks @ravicodder for the improvement.
Tried and works well. 

If it is not too difficult, it might be better to have a simple testcase for searching title at [LuceneSearchTest.java](https://github.com/apache/incubator-zeppelin/blob/master/zeppelin-zengine/src/test/java/org/apache/zeppelin/search/LuceneSearchTest.java).

**Comment by bzz** [comment]
👍  for adding a new test-case for the search API on a new behavoir

**Comment by ravicodder** [comment]
@Leemoonsoo @bzz Thanks for review , Fixed indentation and added test.
@bzz thanks for informing , will take care in future PR's

**Comment by Leemoonsoo** [comment]
Looks good to me!

File: zeppelin-web/src/app/search/result-list.controller.js [hunk 2]
```diff
@@ -74,12 +74,21 @@ angular
         };
       }
 
-      var lines = note.snippet
+      var result = '';
+      if (note.header !== '') {
+        result = note.header + '\n\n' + note.snippet;
+      }
+      else {
```
Comment (bzz) [inline zeppelin-web/src/app/search/result-list.controller.js]: To be consistent with the [rest of the project](https://github.com/ravicodder/incubator-zeppelin/blob/searchTitle/zeppelin-web/src/app/interpreter/interpreter.controller.js#L67) we better move `else` one line up gere and below in this file

File: zeppelin-zengine/src/main/java/org/apache/zeppelin/search/LuceneSearch.java [hunk 3]
```diff
@@ -121,7 +125,7 @@ public LuceneSearch() {
   }
 
   private List<Map<String, String>> doSearch(IndexSearcher searcher, Query query,
-      Analyzer analyzer, Highlighter highlighter) {
+                                             Analyzer analyzer, Highlighter highlighter) {
```
Comment (bzz) [inline zeppelin-zengine/src/main/java/org/apache/zeppelin/search/LuceneSearch.java]: I think previous indentation was correct, according to project style, see [rationale here](https://google.github.io/styleguide/javaguide.html#s4.6.3-horizontal-alignment)

**Comment by bzz** [comment]
@ravicodder thank you for prompt responce!

Looks great to me, modulo style nitpicks above. Let's merge as soon as those are addressed!

**Comment by ravicodder** [comment]
@bzz  Done required changes. Please have a look.

**Comment by prabhjyotsingh** [comment]
LGTM

**Comment by prabhjyotsingh** [comment]
Merging if no more discussion.

**Pull request closed without merging**